# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: 2bf4f54a3077
# Full Hash: 2bf4f54a307746331f89e76ea9be8daaf9732886
# Author: stransky <stransky@redhat.com>
# Date: 2024-12-17 05:09:10
# Regressor Bug: 1934497
# File Overlap Count: 1
# Description:
#   Bug 1934497 [Wayland] Update GtkCompositorWidget for WaylandSurface r=emilio
#   
#   - Replace MozContainerSurfaceLock with WaylandSurfaceLock
#   - Temporary disable GtkCompositorWidget::GetNativeLayerRoot()
#   
# ==============================================================================

diff -r 87613cc32aa4 -r 2bf4f54a3077 widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Mon Dec 16 20:12:16 2024 +0000
+++ b/widget/gtk/GtkCompositorWidget.cpp	Mon Dec 16 20:12:17 2024 +0000
@@ -150,14 +150,18 @@
 #ifdef MOZ_WAYLAND
 RefPtr<mozilla::layers::NativeLayerRoot>
 GtkCompositorWidget::GetNativeLayerRoot() {
+#  if 0
   if (gfx::gfxVars::UseWebRenderCompositor()) {
     if (!mNativeLayerRoot) {
+      LOG("GtkCompositorWidget::GetNativeLayerRoot [%p] create\n",
+          (void*)mWidget.get());
       MOZ_ASSERT(mWidget && mWidget->GetMozContainer());
-      mNativeLayerRoot = layers::NativeLayerRootWayland::CreateForMozContainer(
-          mWidget->GetMozContainer());
+      mNativeLayerRoot = layers::NativeLayerRootWayland::Create(
+          MOZ_WL_SURFACE(mWidget->GetMozContainer()));
     }
     return mNativeLayerRoot;
   }
+#  endif
   return nullptr;
 }
 #endif
@@ -208,7 +212,7 @@
 }
 #endif
 
-UniquePtr<MozContainerSurfaceLock> GtkCompositorWidget::LockSurface() {
+UniquePtr<WaylandSurfaceLock> GtkCompositorWidget::LockSurface() {
   return mWidget ? mWidget->LockSurface() : nullptr;
 }
 